app/frontend/src/features/profile/edit/EditProfile.tsx

```diff
@@ -151,11 +153,30 @@ export default function EditProfileForm() {
     register("radius");
   }, [register]);
 
-  const onSubmit = handleSubmit((data) => {
-    resetUpdate();
-    updateUserProfile({ profileData: data, setMutationError: setErrorMessage });
-    window.scroll({ top: 0 });
-  });
+  const onSubmit = handleSubmit(
+    (data) => {
+      resetUpdate();
+      updateUserProfile(
+        {
+          profileData: data,
+          setMutationError: setErrorMessage,
+        },
+        {
+          // Scoll to top on submission error
+          onError: () => {
+            window.scroll({ top: 0 });
+          },
+        }
+      );
+    },
+    () => {
+      // All field validation errors should scroll to their respective field
+      // Except the avatar, so this scrolls to top on avatar validation error
+      if (errors.avatarKey) {
+        window.scroll({ top: 0 });
```

Comment:
Maybe this one is good to check with design too... but I wonder if it'd be easier/better to use snackbars instead for the error state too so we don't need to litter these scroll ups all over our components